mmengine/runner/log_processor.py

```diff
@@ -339,10 +397,17 @@ def _collect_non_scalars(self, runner, mode: str) -> dict:
         # extract log info and remove prefix to `mode_infos` according to mode.
         for prefix_key, value in infos.items():
             if prefix_key.startswith(mode):
-                key = prefix_key.partition('/')[-1]
+                if self.log_with_hierarchy:
+                    key = prefix_key
+                else:
+                    key = self._remove_prefix(mode, prefix_key)
                 mode_infos[key] = value
         return mode_infos
 
+    def _remove_prefix(self, mode: str, key: str):
+        """Remove the prefix ``train``, ``val`` and ``test`` of the key."""
+        return re.sub(f'{mode}/(.*)', r'\1', key)
```

Comment:
Can it replace with `key.lstrip(f'{mode}/')`?

Reply:
I think `re.sub` will be more robust since `lstrip` will stripe all `mode/` in the string rather than only strip one.

Reply:
```python
def _remove_prefix(self, string: str, prefix: str):
    if string.startswith(f'{prefix}/'):
        return string[len(prefix)+1:]
    else:
        
```